connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConfig.java

```diff
@@ -166,6 +173,29 @@ Duration consumerPollTimeout() {
         return Duration.ofMillis(getLong(CONSUMER_POLL_TIMEOUT_MILLIS));
     }
 
+    public List<ConfigValue> validate() {
+        List<ConfigValue> invalidConfigs = new ArrayList<>();
+
+        if (!this.getBoolean(EMIT_CHECKPOINTS_ENABLED) && !this.getBoolean(SYNC_GROUP_OFFSETS_ENABLED)) {
+            Arrays.asList(new ConfigValue(SYNC_GROUP_OFFSETS_ENABLED), new ConfigValue(EMIT_CHECKPOINTS_ENABLED))
+                    .forEach(configValue -> {
+                        configValue.addErrorMessage("MirrorCheckpointConnector can't run without both " + SYNC_GROUP_OFFSETS_ENABLED + ", " +
+                                EMIT_CHECKPOINTS_ENABLED + " set to false");
+                        invalidConfigs.add(configValue);
+                    });
+        }
```

Comment:
I see the motivation here, but won't it become impossible for people to safely disable the checkpoint connector? It looks like the connector is failing validation in the new `DedicatedMirrorIntegrationTest::testClusterWithEmitOffsetDisabled` case, even though the test itself passes.

Maybe we can skip validation if the connector is disabled (i.e., `getBoolean(ENABLED)` returns `false`)?

Should we add a check

Reply:
Added logic to skip validation if connector is disabled